bioblend/galaxy/invocations/__init__.py

```diff
@@ -425,19 +427,51 @@ def _wait_for_short_term_storage(
         :type interval: float
         :param interval: Time (in seconds) to wait between 2 consecutive checks.
 
-        :rtype: dict
-        :return: The short term storage request.
+        :return: The decoded response if ``json`` is set to ``True``, otherwise
+          the response object
         """
         url = f"{self.gi.url}/short_term_storage/{storage_request_id}"
         is_ready_url = f"{url}/ready"
 
-        def check_and_get_short_term_storage() -> dict[str, Any]:
+        def check_and_get_short_term_storage() -> Any:
             if self._get(url=is_ready_url):
-                return self._get(url=url)
+                return self._get(url=url, json=json, stream=stream)
             raise NotReady(f"Storage request {storage_request_id} is not ready")
 
         return wait_on(check_and_get_short_term_storage, maxwait=maxwait, interval=interval)
 
+    def get_invocation_archive(
+        self, invocation_id: str, payload: dict[str, Any], maxwait: float = 1200
```

Comment:
Can you please replace payload with the parameters you want to expose, and then construct the payload inside the method?

Reply:
done